pkg/collector/corechecks/containers/docker/unbundled_events.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+// +build docker
+
+package docker
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func newUnbundledTransformer(hostname string, types []string) eventTransformer {
+	collectedEventTypes := make(map[string]struct{}, len(types))
+	for _, t := range types {
+		collectedEventTypes[t] = struct{}{}
+	}
+
+	return &unbundledTransformer{
+		hostname:            hostname,
+		collectedEventTypes: collectedEventTypes,
+	}
+}
+
+type unbundledTransformer struct {
+	hostname            string
+	collectedEventTypes map[string]struct{}
+}
+
+func (t *unbundledTransformer) Transform(events []*docker.ContainerEvent) ([]metrics.Event, []error) {
+	var (
+		datadogEvs []metrics.Event
+		errors     []error
+	)
+
+	for _, ev := range events {
+		if _, ok := t.collectedEventTypes[ev.Action]; !ok {
+			continue
+		}
+
+		alertType := metrics.EventAlertTypeInfo
+		if ev.Action == "oom" || ev.Action == "kill" {
+			alertType = metrics.EventAlertTypeError
+		}
+
+		tags, err := tagger.Tag(
+			containers.BuildTaggerEntityName(ev.ContainerID),
+			collectors.HighCardinality,
+		)
+		if err != nil {
+			log.Debugf("no tags for container %q: %s", ev.ContainerID, err)
+		}
+
+		tags = append(tags, fmt.Sprintf("event_type:%s", ev.Action))
+
+		datadogEvs = append(datadogEvs, metrics.Event{
+			Title:          fmt.Sprintf("Container %s: %s", ev.ContainerID, ev.Action),
+			Text:           fmt.Sprintf("Container %s (running image %q): %s", ev.ContainerID, ev.ImageName, ev.Action),
+			Tags:           tags,
+			Priority:       metrics.EventPriorityNormal,
+			Host:           t.hostname,
+			SourceTypeName: dockerCheckName,
+			EventType:      dockerCheckName,
```

Comment:
I don't know how we can do it. But it would have been nice to have a "generic" "EventType" "container". 
For all the container runtimes

Reply:
We can just change the name here, I suppose, and coordinate with the events team to see if there's anything to be done on their side. That's a breaking change for customers though.

Reply:
IMO it is not a breaking change since it is only if they enable the option `unbundle`. 
My final vision is to remove the needs of having one integration by container runtime and have only one integration: `containers`. 
Maybe this change is not necessary in agent 7, since user are used to configure the specific container integrations.